core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1936,25 +1945,38 @@ class KafkaApis(val requestChannel: RequestChannel,
          else
            toDelete += topic.name
       }
-      // If no authorized topics return immediately
-      if (toDelete.isEmpty)
-        sendResponseCallback(results)
-      else {
-        def handleDeleteTopicsResults(errors: Map[String, Errors]): Unit = {
-          errors.foreach {
-            case (topicName, error) =>
-              results.find(topicName)
-                .setErrorCode(error.code)
+      if (!controller.isActive) {
```

Comment:
Same as above. I think it's a little more natural to roll the controller check into the loop above. If you do that, then the structure here will match what I suggested for create topic.